user-guide/Advanced_Modules/User Defined APIs/UD_APIs_UserDefinableApiEndpoint.md

```diff
@@ -0,0 +1,175 @@
+
+1. Open `Internet Information Services (IIS) Manager`.
+
+1. In the *Connections* pane on the left, expand the top node and *Sites* node until you see *Default Web Site*.
+
+1. Select *Default Web Site* and then double-click *URL Rewrite* on the right.
+
+   ![IIS Manager 1](~/user-guide/images/UDAPIS_IIS1.jpg)
+
+1. Select the *Reroute User Definable APIs* rewrite rule and click *Edit* in the pane on the right.
+
+   ![IIS Manager 2](~/user-guide/images/UDAPIS_IIS2.jpg)
+
+1. Under *Action*, in the *Rewrite URL* box, change the default port (5002) to the port you want to use.
+
+   ![IIS Manager 3](~/user-guide/images/UDAPIS_IIS3.jpg)
+
+1. Click *Apply*.
+
+### Serilog
+
+Serilog is the logging service used for *UserDefinableApiEndpoint*. Here you can change where the log files should be located, how big they can get, and how many files should be kept. You can also change the default log levels.
+
+By default, a log file has a maximum size of 5 MB, and at most 3 files will be kept. When there are 3 files already and a new file needs to be created, the oldest file will be removed.
+
+For example, this is the default configuration:
+
+```json
+{
+  ...
+  "Serilog": {
+    "Using": [ "Serilog.Sinks.Console", "Serilog.Sinks.File" ],
+    "MinimumLevel": {
+      "Default": "Information",
+      "Override": {
+        "Default": "Debug",
+        "System": "Information",
+        "Microsoft": "Information"
+      }
+    },
+    "WriteTo": [
+      {
+        "Name": "Console"
+      },
+      {
+        "Name": "File",
+        "Args": {
+          "path": "%ProgramData%\\Skyline Communications\\DataMiner UserDefinableApiEndpoint\\Logs\\UserDefinableApiEndpoint.txt",
+          "fileSizeLimitBytes": 5000000,
+          "rollOnFileSizeLimit": true,
+          "retainedFileCountLimit": 3
+        }
+      }
+    ],
+    "Enrich": [ "FromLogContext" ]
+  },
+  ...
+}
+```
+
+### UserDefinableAPIs
+
+This section contains options specific to the DataMiner User-Definable APIs module:
+
+- **NatsSubject**: The subject used for internal NATS messaging. This should not be changed.
+
+- **SessionConfigPath**: The path to the *SLCLoud.xml* file.
+
+- **CredentialsConfigPath**: The path to the credentials used to connect to DataMiner over NATS.
+
+- **MessageBrokerTimeOutSeconds**: The time the message broker sending the NATS trigger to SLNet will wait for a response before it times out. By default, this is set to 300 seconds (i.e. 5 minutes).
+
+For example, this is the default configuration:
+
+```json
+{
+  ...
+  "UserDefinableAPIs": {
+    "NatsSubject": "Skyline.DataMiner.Protobuf.Apps.UserDefinableApis.Api.v1.UserDefinableApiTriggerRequest",
+    "SessionConfigPath": "C:\\Skyline DataMiner\\SLCloud.xml",
+    "CredentialsConfigPath": "C:\\Skyline DataMiner\\NATS\\nsc\\.nkeys\\creds\\DataMinerOperator\\DataMinerAccount\\DataMinerUser.creds",
+    "MessageBrokerTimeOutSeconds": 300
+  }
+}
+```
+
+## HTTP(S) support
```

Comment:
It's not recommended to allow API requests over HTTP, that's why I'm not a fan of adding information of how to do it. The firewall depends on what firewall is used, we could add an explanation on how to add this for Windows firewall, but in case a customer has a physical firewall, it's up to them to change the configuration of that to allow the incoming requests. I agree with moving this section up and changing the title.